app/controllers/incomeSources/manage/ManageObligationsController.scala

```diff
@@ -0,0 +1,222 @@
+            addedBusiness <- user.incomeSources.businesses.find(x => x.incomeSourceId.contains(incomeSourceId))
+            businessName <- addedBusiness.tradingName
+          } yield (addedBusiness, businessName)
+          businessDetailsParams match {
+            case Some((_, name)) => name
+            case None => "Not Found"
+          }
+        }
+        else {
+          if (mode == "UK") "UK property"
+          else "Foreign property"
+        }
+
+        val (year1,year2) = taxYear.splitAt(4)
+        val currenttaxYearStart = dateService.getCurrentTaxYearEnd() - 1
+        if ((year1.toInt == currenttaxYearStart && year2.drop(1).toInt == currenttaxYearStart+1)
+          || (year1.toInt == currenttaxYearStart+1 && year2.drop(1).toInt == currenttaxYearStart+2)){
+          if (changeTo == "annual" || changeTo == "quarterly") {
+            getIncomeSourceId(mode, incomeSourceId) match {
+              case Left(error) => Logger("application").error(
+                s"[BusinessAddedObligationsController][handleRequest] - ${error.getMessage}")
+                if (isAgent) Future(itvcErrorHandlerAgent.showInternalServerError())
+                else Future(itvcErrorHandler.showInternalServerError())
+              case Right(value) =>
+                nextUpdatesService.getObligationsViewModel(value, showPreviousTaxYears = false) map { viewModel =>
+                  if (isAgent) Ok(obligationsView(viewModel, mode, addedBusinessName, taxYear, changeTo, isAgent, backUrl, postUrl))
+                  else Ok(obligationsView(viewModel, mode, addedBusinessName, taxYear, changeTo, isAgent, backUrl, postUrl))
+                }
+            }
+          }
+          else{
+            Logger("application").error(
+              s"[BusinessAddedObligationsController][handleRequest] - invalid change to mode provided")
+            if (isAgent) Future(itvcErrorHandlerAgent.showInternalServerError())
+            else Future(itvcErrorHandler.showInternalServerError())
+          }
+        }
+        else{
+          Logger("application").error(
+            s"[BusinessAddedObligationsController][handleRequest] - invalid tax year provided")
+          if (isAgent) Future(itvcErrorHandlerAgent.showInternalServerError())
+          else Future(itvcErrorHandler.showInternalServerError())
+        }
+      }
+    }
+  }
+
+  def getIncomeSourceId(mode: String, id: String)(implicit user: MtdItUser[_]): Either[Throwable, String] = {
+    mode match {
+      case "SE" => Right(id)
+      case "UK" => user.incomeSources.properties.find(x => x.isUkProperty) match {
+        case Some(value) => Right(value.incomeSourceId.getOrElse(""))
+        case None => Left(new Error("Failed to find incomeSource Id"))
+      }
+      case "FP" => user.incomeSources.properties.find(x => x.isForeignProperty) match {
+        case Some(value) => Right(value.incomeSourceId.getOrElse(""))
+        case None => Left(new Error("Failed to find incomeSource Id"))
```

Comment:
pls add incomSourceId details to the error as this would help to investigate this error

Reply:
If we are in mode "UK" or "FP" (so either of the property pages) then no incomeSourceId is supplied so I cannot output it